src/app/api/v1/user/breaches/bulk-resolve/route.ts

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { authOptions } from "../../../../utils/auth";
+import { getServerSession } from "next-auth";
+import { logger } from "../../../../../functions/server/logging";
+import {
+  BreachBulkResolutionRequest,
+  Subscriber,
+} from "../../../../../(nextjs_migration)/(authenticated)/user/breaches/breaches.js";
+import { getBreaches } from "../../../../../functions/server/getBreaches";
+import { getAllEmailsAndBreaches } from "../../../../../../utils/breaches";
+import {
+  getSubscriberByEmail,
+  setBreachResolution,
+} from "../../../../../../db/tables/subscribers";
+
+export async function PUT(req: NextRequest): Promise<NextResponse> {
+  const session = await getServerSession(authOptions);
+  if (!session?.user?.subscriber || typeof session.user?.email !== "string") {
+    return new NextResponse(
+      JSON.stringify({ success: false, message: "Unauthenticated" }),
+      { status: 401 },
+    );
+  }
+
+  try {
+    const subscriber: Subscriber = await getSubscriberByEmail(
+      session?.user?.email,
```

Comment:
Should this just be `session.user.email`?
IIUC, we know `session.user` should be valid since we would have already bounced unauthenticated folks above on L23-26.

Reply:
Yes, that would work as well — see the reply to [your other comment](https://github.com/mozilla/blurts-server/pull/3780#discussion_r1403255773).